File: apps/api/src/modules/audit-log/audit-log.controller.ts
import { Controller, Get, Query, UseGuards } from '@nestjs/common';
import { ApiTags, ApiOperation, ApiBearerAuth } from '@nestjs/swagger';
import { AuthGuard } from '@nestjs/passport';
import { SuperAdminGuard } from '../../common/guards';
import { AuditLogService } from './audit-log.service';

@ApiTags('Super Admin - Audit Logs')
@ApiBearerAuth()
@UseGuards(AuthGuard('jwt'), SuperAdminGuard)
@Controller('super-admin/audit-logs')
export class AuditLogController {
  constructor(private readonly auditLogService: AuditLogService) {}

  @Get()
  @ApiOperation({ summary: 'List audit logs with filters and pagination' })
  findAll(
    @Query('action') action?: string,
    @Query('entity_type') entity_type?: string,
    @Query('user_id') user_id?: string,
    @Query('user_email') user_email?: string,
    @Query('from') from?: string,
    @Query('to') to?: string,
    @Query('page') page?: string,
    @Query('limit') limit?: string,
  ) {
    return this.auditLogService.findAll({
      action,
      entity_type,
      user_id,
      user_email,
      from,
      to,
      page: page ? parseInt(page, 10) : undefined,
      limit: limit ? parseInt(limit, 10) : undefined,
    });
  }

  @Get('stats')
  @ApiOperation({ summary: 'Get audit log statistics for last 30 days' })
  getStats() {
    return this.auditLogService.getStats();
  }
}

File: apps/api-java/src/main/java/br/com/menufacil/controller/AuditLogController.java
package br.com.menufacil.controller;

import br.com.menufacil.config.security.RequirePermissions;
import br.com.menufacil.dto.AuditLogResponse;
import br.com.menufacil.dto.AuditLogStatsResponse;
import br.com.menufacil.service.AuditLogService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;
import java.util.UUID;

@Tag(name = "Super Admin - Audit Logs", description = "Consulta de logs de auditoria do sistema")
@RestController
@RequestMapping("/super-admin/audit-logs")
@RequiredArgsConstructor
public class AuditLogController {

    private final AuditLogService auditLogService;

    @Operation(summary = "Listar logs de auditoria com filtros e paginação")
    @RequirePermissions("report:read")
    @GetMapping
    public ResponseEntity<Page<AuditLogResponse>> findAll(
            @RequestParam(required = false) String action,
            @RequestParam(name = "entityType", required = false) String entityType,
            @RequestParam(name = "userId", required = false) UUID userId,
            @RequestParam(name = "userEmail", required = false) String userEmail,
            @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime from,
            @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime to,
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "50") int limit) {

        return ResponseEntity.ok(
                auditLogService.findAll(action, entityType, userId, userEmail, from, to, page, limit));
    }

    @Operation(summary = "Estatísticas agregadas dos logs de auditoria (últimos 30 dias)")
    @RequirePermissions("report:read")
    @GetMapping("/stats")
    public ResponseEntity<AuditLogStatsResponse> getStats() {
        return ResponseEntity.ok(auditLogService.getStats());
    }
}
